src/Database/Driver/Mysql.php

```diff
@@ -290,8 +290,8 @@ public function version(): string
 
             if (strpos($this->_version, 'MariaDB') !== false) {
                 $this->serverType = static::SERVER_TYPE_MARIADB;
-                $parts = explode(':', $this->_version);
-                $this->_version = array_pop($parts);
+                preg_match('/^(?:5\.5\.5-)?(\d+\.\d+\.\d+.*-MariaDB[^:]*)/', $this->_version, $matches);
```

Comment:
This will be difficult to test. Users will just have to report issues with specific builds.

Reply:
We can update it as and when needed if users report a string which isn't covered.

Reply:
Should we have a testable way to access this so that we can add build names to the test suite as we collect them? Without that we'll have to hope that adding new names won't break the old ones.

Reply:
I have added tests for `version()` method now.